service/storage_proxy.cc

```diff
@@ -4610,6 +4612,45 @@ future<bool> storage_proxy::cas(schema_ptr schema, shared_ptr<cas_request> reque
     co_return condition_met;
 }
 
+bool storage_proxy::is_likely_to_respond_in_time(const gms::inet_address& ep, clock_type::time_point timeout) {
+    clock_type::time_point now = clock_type::now();
+    if (timeout <= now) {
+        return false;
+    }
+    if (fbu::is_me(ep)) {
+        return true;
+    }
```

Comment:
I could be overloaded.

Reply:
I never reached such a scenario in my tests, but I think that if the coordinator node is already overloaded, it should either start returning `OverloadedExeption` to clients or creating backpressure with the admission semaphore, and ideally never even reach a point when it needs to exclude itself as a replica in order to serve other requests. Still, maybe the assumption should be revisited, I'll try to figure out how to reach such scenario (one obvious way would be to use a coordinator with really slow hard drives).

Reply:
It could be overloaded as a replica but not as a coordinator (disk is overloaded but not cpu).

Reply:
We can just drop this check, no? And gather the info for is_me().